core/services/relay/evm/write_target.go

```diff
@@ -69,9 +69,14 @@ func NewWriteTarget(ctx context.Context, relayer *Relayer, chain legacyevm.Chain
 			},
 		},
 	}
-
 	chainWriterConfig.MaxGasPrice = chain.Config().EVM().GasEstimator().PriceMax()
-	cw, err := NewChainWriterService(lggr.Named("ChainWriter"), chain.Client(), chain.TxManager(), chain.GasEstimator(), chainWriterConfig)
+
+	encodedWriterConfig, err := json.Marshal(chainWriterConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal chainwriter config: %w", err)
+	}
+
+	cw, err := relayer.NewChainWriter(ctx, encodedWriterConfig)
```

Comment:
Does it make sense to add tests for this in `write_target_test.go`?

Reply:
I'm not sure what I would test here. I'm just replacing the constructor that it's using.